src/drivers/kinetis/adc/adc.cpp

```diff
@@ -207,72 +185,86 @@ ADC::~ADC()
 	irqstate_t flags = px4_enter_critical_section();
 	_REG(KINETIS_SIM_SCGC3) &= ~SIM_SCGC3_ADC1;
 	px4_leave_critical_section(flags);
+
+	perf_free(_sample_perf);
 }
 
-int
-ADC::init()
+int board_adc_init()
 {
-	/* Input is Buss Clock 56 Mhz We will use /8 for 7 Mhz */
+	static bool once = false;
 
-	irqstate_t flags = px4_enter_critical_section();
+	if (!once) {
```

Comment:
This is Not a once.  It is not correct if it does not set once true

This HW does not support versioning and does not need the once.

Reply:
Dropped.